### PR TITLE
Makefile for docker images. Docker with cuda and cudnn

### PR DESCRIPTION
**Pull request by Sette**
### Which issue(s) does this Pull Request fix?

resolves #000

### CODE REVIEW

**Comment by gabrieldemarmiesse** [comment]
Could we just target tf.keras? It would make things much simpler. The multi-backend keras is deprecated.

**Comment by Sette** [comment]
You're right. I will improve the dockerfile and remove the keras multi-beckend.

File: docker/README.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+Build the container and start a bash
+
+    $ make bash
+
+For GPU support install NVIDIA drivers (ideally latest) and
+[nvidia-docker](https://github.com/NVIDIA/nvidia-docker). Run using
+
+    $ make notebook GPU=0 # or [ipython, bash]
+
+Switch between Theano and TensorFlow
```
Comment (jhfjhfj1) [inline docker/README.md]: We are not supporting Theano backend for AutoKeras.

File: docker/README.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+    $ make notebook BACKEND=theano
```
Comment (jhfjhfj1) [inline docker/README.md]: Same here.

File: docker/README.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+Mount a volume for external data sets
+
+    $ make DATA=~/mydata
+
+Prints all make tasks
+
+    $ make help
+
+You can change Theano parameters by editing `/docker/theanorc`.
```
Comment (jhfjhfj1) [inline docker/README.md]: Same here.

**Comment by haifeng-jin** [comment]
I am merging it.
I haven't test it yet, but it definitely more complete than the previous file.
We can fix it later if anyone finds any bug.

**Comment by gabrieldemarmiesse** [comment]
@Sette tensorflow has ready to use docker images, with gpu enable and jupyter included. Do you think we could use that to reduce the size of the Dockerfile? 

https://hub.docker.com/r/tensorflow/tensorflow

**Comment by haifeng-jin** [comment]
@gabrieldemarmiesse When we make sure the docker file is good, we may try the GitHub Package feature.
https://github.com/keras-team/autokeras/packages?package_type=Docker
What do you think?

**Comment by gabrieldemarmiesse** [comment]
@haifeng-jin , you mean to host the produced docker image? We have the choice between GitHub package and DockerHub on this.

From [this thread](https://github.community/t5/GitHub-Actions/Use-docker-images-from-GitHub-Package-Registry/td-p/30407) it seems that dockerhub is more integrated into the docker cli than GitHub packages. (not a big surprise since it's the same company making both).

It seems that unless we have a strong reason not to, it would be preferable to push the images to DockerHub.

I hope I didn't misinterpret your question.

**Comment by haifeng-jin** [comment]
@gabrieldemarmiesse Great! Then let's just use DockerHub. We may automate the push process with GitHub Actions.

**Comment by gabrieldemarmiesse** [comment]
Ok sure thing. Beforehand, we should rewrite the Dockerfile to use the official Tensorflow image rather than the nvidia/cuda one. It's going to be less maintenance in the long run. I can work on that when I have the time.

**Comment by haifeng-jin** [comment]
@gabrieldemarmiesse Awesome! Thank you

**Pull request merged**